Sources/Basics/CMakeLists.txt

```diff
@@ -70,6 +70,9 @@ target_link_libraries(Basics PUBLIC
 target_link_libraries(Basics PRIVATE
   SPMSQLite3
   TSCclibc)
+
+target_compile_options(Basics PRIVATE -package-name SwiftPM)
```

Comment:
Does it make sense to hoist this to the top level and change it to:

```
add_compile_options($<$<COMPILE_LANGUAGE:Swift>:-package-name SwiftPM>)
```

Reply:
Good shout, applying now.

Reply:
This gets applied in the end as `$<l:-package-name SwiftPM>"` verbatim somehow. But 
using `set(CMAKE_Swift_FLAGS "${CMAKE_Swift_FLAGS} -package-name SwiftPM")` instead works now, hope that's good enough.

Reply:
Hmm, that seems odd.  I think that we might have uncovered a bug in CMake!  Thanks for the heads up. I think that the workaround is okay for now, but we should _really_ limit it to the version ASAP. This is not meant to be touched by the package, it is meant to be user controlled only.

Reply:
Bleh, you just need to write this verbosely :'(

add_compile_options($<$<COMPILE_LANGUAGE:Swift>:-package-name> $<$<COMPILE_LANGUAGE:Swift>:SwiftPM>)

Reply:
> `add_compile_options($<$<COMPILE_LANGUAGE:Swift>:-package-name SwiftPM>)`

Generator expressions don't do lists. You'll need to split them. 
`add_compile_options($<$<COMPILE_LANGUAGE:Swift>:-package-name> $<$<COMPILE_LANGUAGE:Swift>:SwiftPM>)`

Reply:
> we should _really_ limit it to the version ASAP
@compnerd are you referring to the Swift version or something else? I'm leaning towards using `add_compile_options($<$<COMPILE_LANGUAGE:Swift>:-package-name$<SEMICOLON>SwiftPM>)` in the end, would that be sufficient as is?

Reply:
No preference from me. They both get the job done. Just throwing out ideas. I guess there's one more option, quotes.
`add_compile_options("$<$<COMPILE_LANGUAGE:Swift>:-package-name;SwiftPM>")` should also work.

Reply:
When we were thinking it was a CMake bug and modifying `CMAKE_Swift_FLAGS`, I meant limiting that to the impacted CMake version. It ended up being a silly thing about how the generator expression works.  Now that we have changed that to the form that @etcwilde quoted, I think that this issue is resolved.  Now the question is, when can I have this? 😆